src/python_inspector/resolution.py

```diff
@@ -178,7 +308,7 @@ def _iter_matches(self, identifier, requirements, incompatibilities):
         """
         Yield candidates for the given identifier, requirements and incompatibilities
         """
-        name, _, _extras = identifier.partition("[")
+        name, _, _ = identifier.partition("[")
```

Comment:
Add some comment to explain what and why this is done here. I get this may be about extras... but that's not obvious. Make sure you have a test for this